### ISSUE TITLE
Please make quoting of cookie value optional.

### ISSUE THREAD

**Issue by jannic** [AsyncHttpClient/async-http-client#425]
I have an issue with a web service I need to connect to, which doesn't like quotes around cookie values.

As async-http-client adds quotes automatically if the value contains some special characters (as defined in AsyncHttpProviderUtils.add(sb, name, val)), I can't use AsyncHttpProviderUtils.encodeCookies. And as dispatch (https://github.com/dispatch/reboot) uses this method in its cookie code, I can't use that either.

Please note that I don't know if this is a bug - the RFCs are not really clear and it's not even clear which RFC would apply, here. According to RFC 6265, the quotes are part of 'cookie-value' as defined in 4.1.1. And in the sections describing how to parse the cookie headers, it's specified how to remove spaces etc., but I didn't find a single word about removing quotes. That could imply that just adding quotes to the cookie-value changes that value - which would be a bug. On the other hand, well, these are quotes. They are used for a reason, and it wouldn't make much sense if using them would change the meaning of a cookie.

TL;DR: Specification is ambiguous, and there are servers incompatible with the current behavior, so an option to change that behavior would be nice.
